src/Components/DataHistoryExplorer/DataHistoryExplorer.tsx

```diff
@@ -39,24 +80,371 @@ const DataHistoryExplorer: React.FC<IDataHistoryExplorerProps> = (props) => {
+            );
+            dispatch({
+                type: DataHistoryExplorerActionType.REMOVE_TIME_SERIES_TWINS,
+                payload: { seriesIdx: selectedIdx }
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .RemoveSeries;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.itemIndex]: selectedIdx
+                }
+            ]);
+            /**
+             * when a series is removed, also remove if from usedColors reference
+             * so that whenever adding a new series we can use these available colors first
+             * before picking the next color in Highcharts palette
+             */
+            usedSeriesColorsRef.current.splice(
+                usedSeriesColorsRef.current.findIndex(
+                    (c) =>
+                        c ===
+                        state.timeSeriesTwins.find(
+                            (tsTwin) => tsTwin.seriesId === seriesId
+                        ).chartProps.color
+                ),
+                1
+            );
+        },
+        [state.timeSeriesTwins, dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnViewModeChange = useCallback(
+        (viewerMode: TimeSeriesViewerMode) => {
+            dispatch({
+                type: DataHistoryExplorerActionType.SET_VIEWER_MODE,
+                payload: { viewerMode: viewerMode }
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction.ChangeView;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.view]: viewerMode
+                }
+            ]);
+        },
+        [dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnChartOptionChange = useCallback(
+        (chartOptions: ITimeSeriesCommandBarOptions) => {
+            dispatch({
+                type:
+                    DataHistoryExplorerActionType.SET_COMMAND_BAR_CHART_OPTIONS,
+                payload: { chartOptions: chartOptions }
+            });
+            logDebugConsole(
+                'debug',
+                `Chart options changed: ${JSON.stringify(chartOptions)}`
+            );
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .ChangeChartOption;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.chartOptions]: chartOptions
+                }
+            ]);
+        },
+        [
+            state.timeSeriesTwins,
+            dispatch,
+            logDebugConsole,
+            sendDataHistoryExplorerUserTelemetry
+        ]
+    );
+
+    const handleOnDownloadTableClick = useCallback(() => {
+        downloadText(
```

Comment:
nit: this helper is downloading the file as a CSV  but you are serializing the data to JSON. Not sure of the implications there. Just wanted to call it out.

Reply:
good catch, I think the mime type of csv is required to have the same keys across all items, and since we didn't have any different keys before it was all good. but to be more accurate I changed it to application/json and changed the method name to downloadJSON since we all use .json format to download files across the cardboard now.